packages/loader/driver-definitions/src/urlResolver.ts

```diff
@@ -38,6 +38,20 @@ export interface IUrlResolver {
     ): Promise<string>;
 }
 
+// Information that can be returned by a lightweight, seperately exported driver function. Used to preanalyze a URL for
+// driver compatibility and preload information.
+export interface DriverPreCheckInfo {
```

Comment:
I'd rather this not be in driver definitions, as i'm not sure it is generally applicable to all drivers. I think we should just make this odsp specific for now. once we have more servers we can see if there is a common pattern here

Reply:
I could see preconnectDomains not necessarily being generic for all drivers. 

However, not sure I agree about componentPath and containerCodeHint, this seems like it would benefit all drivers. Would these not be useful in routerlicious for example?

Reply:
Yes, I put it here precisely because I think containerCodeHint/preconnectDomains could be useful for other drivers. We already have the spo specific workaround (resolveDataStore, the method that's being removed) and this PR is exactly to take that odsp specific implementation and turn it into a recommendation for all drivers to implement, with an eye towards actually implementing the same thing for FRS/routerlicious/local-test/peer to peer drivers.

One thing I did miss in my design here though is the full replacement for componentPath should include a boolean for "supports or not", which is a necessary functionality for any heavy driver to be reasonably consumed in other applications and most of the point of #4498

Reply:
ok. then we'll have to do a bit more to generalize it all